libyul/optimiser/Metrics.cpp

```diff
@@ -180,7 +181,8 @@ void CodeCost::visit(Expression const& _expression)
 
 void CodeCost::addInstructionCost(evmasm::Instruction _instruction)
 {
-	evmasm::Tier gasPriceTier = evmasm::instructionInfo(_instruction).gasPriceTier;
+	EVMDialect const* evmDialect = dynamic_cast<EVMDialect const*>(&m_dialect);
+	evmasm::Tier gasPriceTier = evmasm::instructionInfo(_instruction, evmDialect->evmVersion()).gasPriceTier;
```

Comment:
```suggestion
	evmasm::Tier gasPriceTier = evmasm::instructionInfo(_instruction, evmVersionFromDialect(m_dialect)).gasPriceTier;
```
Since you add that anyways :-)?

Reply:
Indeed. I will fix.